doc/version-control-fields.rst

```diff
@@ -0,0 +1,179 @@
+Version Control System Fields
+=============================
+
+Most of the version control system (VCS) fields types are common to both
+``source-repository`` and ``source-repository-package`` stanzas.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 30 30 40
+
+    * - Field Name
+      - source-repository (head|this)
+      - source-repository-package
+    * - type
+      - 🗹
+      - 🗹
+    * - location
+      - 🗹
+      - 🗹
+    * - branch
+      - 🗹
+      - 🗹
+    * - tag
+      - 🗹
+      - 🗹
+    * - subdir
+      - 🗹 (0 or 1)
+      - 🗹 (0 or 1 for each dependency)
+    * - module (CVS only)
+      - 🗹
+      - ☐
+    * - post-checkout-command
+      - ☐
+      - 🗹
```

Comment:
The symbols look broken on my Linux machine in Firefox.
![version_control_fields](https://github.com/haskell/cabal/assets/5301202/30016a50-8ae5-4369-a417-6cd536a6a067)

Reply:
Renders fine for me on Ubuntu with Firefox. What locale are you using @malteneuss? Is the problem there if you use the link from the CI rendering, from the "docs/readthedocs.org:cabal" job (might need to scroll down to see the link).

Reply:
They're also broken for me in the RTD documentation. Ubuntu 22.04.3, Chrome.

Reply:
Do code blocks like the following with unicode chars render well? These snippets are from `doc/how-to-source-packages.rst`.

```
       (package description)      source-repository         (Git repository)
      solver-benchmarks.cabal ──────────────────────────⮚ github/haskell/cabal

        (project)       source-repository-package         (Git repository)
      cabal.project ──────────────────────────────────⮚ github/haskell/vector
```

Reply:
It appears to have half-rendered: the lines are fine but there's a small block at the end which may have been intended to be an arrowhead?

Reply:
![Screenshot_2024-05-02_13-29-20](https://github.com/haskell/cabal/assets/2036384/d220f326-f5e1-45bd-88c0-5db05d74b0b6)

Reply:
Yes, the lines I grabbed from the output from the `tree` command in ubuntu and the arrow from https://www.amp-what.com/unicode/search/arrowhead.

Reply:
Thanks for the screenshot @geekosaur. Unfortunate that those unicode chars don't work. In the past I've had good results using them in web pages.

Reply:
Then let's just use `---->` ascii arrows and ascii symbols if possible for now. We can try out beatifications later

Reply:
As it is only the arrowheads, I replaced those and the ticked checkboxes.